scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5082 @@
+        :math:`f(x)` by:
+
+        .. math::
+
+            G(a) = \int_a^r f(x) dx
+
+        The two argument version is:
+
+        .. math::
+
+            G(a, b) = \int_l^a f(x) dx + \int_b^r f(x) dx
+
+        The CCDF takes on its minimum value of :math:`0` for :math:`x ≥ r`
+        and its maximum value of :math:`1` for :math:`x ≤ l`.
+
+        Examples
+        --------
+        Instantiate a distribution with the desired parameters:
+
+        >>> import numpy as np
+        >>> from scipy import stats
+        >>> X = stats.Uniform(a=-0.5, b=0.5)
+
+        Evaluate the CCDF at the desired argument:
+
+        >>> X.ccdf(0.25)
+        0.25
+        >>> np.allclose(X.ccdf(0.25), 1-X.cdf(0.25))
+        True
+
+        """  # noqa: E501
+        if y is None:
+            return self._ccdf1(x, method=method)
+        else:
+            return self._ccdf2(x, y, method=method)
+
+    @_cdf2_input_validation
+    def _ccdf2(self, x, y, *, method):
+        return self._ccdf2_dispatch(x, y, method=method, **self._parameters)
+
+    @_dispatch
+    def _ccdf2_dispatch(self, x, y, *, method=None, **kwargs):
+        if self._overrides('_ccdf2_formula'):
+            method = self._ccdf2_formula
+        else:
+            method = self._ccdf2_addition
+        return method
+
+    def _ccdf2_formula(self, x, y, **kwargs):
+        raise NotImplementedError(self._not_implemented)
+
+    def _ccdf2_addition(self, x, y, **kwargs):
+        cdf_x = self._cdf_dispatch(x, **kwargs)
+        ccdf_y = self._ccdf_dispatch(y, **kwargs)
+        # even if x > y, cdf(x, y) + ccdf(x,y) sums to 1
+        return cdf_x + ccdf_y
+
+    @_set_invalid_nan
+    def _ccdf1(self, x, *, method):
+        return self._ccdf_dispatch(x, method=method, **self._parameters)
+
+    @_dispatch
+    def _ccdf_dispatch(self, x, method=None, **kwargs):
+        if self._overrides('_ccdf_formula'):
+            method = self._ccdf_formula
+        elif self._overrides('_logccdf_formula'):
+            method = self._ccdf_logexp
+        elif self._tol is _null and self._overrides('_cdf_formula'):
+            method = self._ccdf_complementarity
+        else:
+            method = self._ccdf_quadrature
+        return method
+
+    def _ccdf_formula(self, x, **kwargs):
+        return NotImplementedError(self._not_implemented)
+
+    def _ccdf_logexp(self, x, **kwargs):
+        return np.exp(self._logccdf_dispatch(x, **kwargs))
+
+    def _ccdf_complementarity(self, x, **kwargs):
+        return 1 - self._cdf_dispatch(x, **kwargs)
+
+    def _ccdf_quadrature(self, x, **kwargs):
+        _, b = self._support(**kwargs)
+        return self._quadrature(self._pdf_dispatch, limits=(x, b),
+                                kwargs=kwargs)
+
+    ## Inverse cumulative distribution functions
+
+    @_set_invalid_nan
+    def ilogcdf(self, x, *, method=None):
+        r"""Inverse of the logarithm of the cumulative distribution function
```

Comment:
The teensiest of nits, here and elsewhere.

```suggestion
        r"""Inverse of the logarithm of the cumulative distribution function.
```

Reply:
No, this sort of comment is fine. I want these to be beyond reproach. 
I always forget that this is what numpydoc shows because, you know, English. It's not a sentence.

Reply:
Ah ok, disregard my comments about grammar. (To my eye, this reads like a senstence because the verb can be omitted in Russian).

Reply:
Fixed.

Reply:
@tupui or @ev-br please confirm that I've put periods at the end of function one-line summaries.